Update federation controllers to use release 1.5 clientset

caesarxuchao opened the pull request:
This PR is based on #32243, we won't merge it until next week, to make cherry-picks easier.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/32246)

<!-- Reviewable:end -->

Commit:
changes in federation controllers

nikhiljindal commented:
cc @kubernetes/sig-cluster-federation

caesarxuchao commented:
@nikhiljindal Rebased, PTAL. Thanks!

nikhiljindal commented on federation/pkg/federation-controller/namespace/namespace_controller.go:
```diff
@@ -22,14 +22,14 @@ import (
 
 	federation_api "k8s.io/kubernetes/federation/apis/federation/v1beta1"
-	federation_release_1_4 "k8s.io/kubernetes/federation/client/clientset_generated/federation_release_1_4"
+	federation_kubeclientset "k8s.io/kubernetes/federation/client/clientset_generated/federation_release_1_5"
```
same

caesarxuchao replied:
I'll rename it. Thanks.

nikhiljindal commented on federation/pkg/federation-controller/ingress/ingress_controller.go:
```diff
@@ -23,14 +23,14 @@ import (
 
 	federation_api "k8s.io/kubernetes/federation/apis/federation/v1beta1"
-	federation_release_1_4 "k8s.io/kubernetes/federation/client/clientset_generated/federation_release_1_4"
+	federation_kubeclientset "k8s.io/kubernetes/federation/client/clientset_generated/federation_release_1_5"
```
At other places, we use federationclientset. Should be consistent.

nikhiljindal commented:
LGTM, thx!

caesarxuchao commented on pkg/client/clientset_generated/release_1_5/typed/core/v1/fake/fake_pod_expansion.go:
```diff
@@ -44,3 +45,14 @@ func (c *FakePods) GetLogs(name string, opts *v1.PodLogOptions) *restclient.Requ
 	return &restclient.Request{}
 }
+
+func (c *FakePods) Evict(eviction *policy.Eviction) error {
```
This is to fix the unit test failure. The function is required by the PodInterface. @nikhiljindal

caesarxuchao commented:
I rebased the code, addressed the comment on the inconsistent import names, fixed a build failure. They are all quite trivial, I added back the lgtm myself, @nikhiljindal please remove it if you see any issue. Thanks!

caesarxuchao commented:
@k8s-bot test this, issue #IGNORE

caesarxuchao commented:
@k8s-bot gke e2e test this, issue #33611
